### PR TITLE
fix: show active DAG runs in dashboard timeline

### LINKED ISSUE #2475: bug: Timeline not showing active manually started workflow
## Describe the bug

Active manually started workflow not showing in 'Overview -> Timeline' although the stats displayed show
that the are active workflows.

## To Reproduce

1. Create a DAG that runs for several minutes/hours
2. Start the workflow manually
3. Once running, check 'Overview -> Timeline' to see that there are active workflows, but timeline not showing it
4. Check 'Overview -> Cockpit' to see that workflow is in the 'running' state

## Expected behavior

Timeline should display workflows that are currently active

## Actual behavior

Timeline is not showing currently active workflows.

## Environment

- Dagu version: 2.11.1
- OS: Ubuntu 24.04\
- Go version: N/A
- Installation method: Dagu Helm chart on k8s (5 nodes, with 5 workers)

## DAG configuration

If applicable, paste the relevant DAG YAML (redact any sensitive information):

```yaml

```

### Worflow running
Image

### Active workflow, but not shown in timeline
Image

### Cockpit shows workflow running
Image

### Workflow shown in timeline after aborted
Image

### PR DESCRIPTION
**Pull request by yohamta0**
## Summary

- show active DAG runs in the dashboard Timeline while they are still running
- retain compatibility with the legacy `"-"` unfinished-time sentinel
- add component-level regression coverage for an empty `finishedAt` value

## Root cause

Running DAG-run summaries represent the unset finish time as an empty string. The Timeline only recognized the legacy `"-"` sentinel, parsed the empty string as an invalid date, and then discarded the run while constructing chart items.

## Impact

Active runs now appear in the Timeline using the current time as the interval end. Completed-run rendering is unchanged.

## Testing

- `pnpm test`
- `pnpm typecheck`
- `pnpm exec eslint src/features/dashboard/components/DashboardTimechart.tsx src/features/dashboard/components/__tests__/DashboardTimechart.test.tsx`
- `pnpm exec prettier --check src/features/dashboard/components/DashboardTimechart.tsx src/features/dashboard/components/__tests__/DashboardTimechart.test.tsx`

Closes #2475

---
## Summary by cubic
Fixes the dashboard Timeline so active DAG runs appear while still running by treating an empty `finishedAt` as ongoing and using the current time as the end. Legacy `'-'` support remains; completed runs are unchanged.

- **Bug Fixes**
  - Treat empty `finishedAt` as active and set the end to now.
  - Keep `'-'` sentinel compatibility and add a component test asserting empty `finishedAt` uses now for the timeline end.

<sup>Written for commit 66b86ff9b8202d12512bff6b333cfeb9cb3b3f79. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/dagucloud/dagu/pull/2477?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved timeline display for running workflows by using the current time when no valid finish time is available.

* **Tests**
  * Added coverage to verify that active workflow runs appear correctly with a valid timeline end time.

### CODE REVIEW

**Comment by yohamta0** [comment]
@coderabbitai review

File: ui/src/features/dashboard/components/__tests__/DashboardTimechart.test.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    const items = timelineState.dataSet?.get() ?? [];
+    expect(items).toHaveLength(1);
+    expect(items[0]).toEqual(
+      expect.objectContaining({
+        id: 'long-running-dag_run-1',
+        content: 'long-running-dag',
+      })
+    );
+    expect(items[0]?.end.getTime()).not.toBeNaN();
+  });
```
[the remark this answers is not in the document]
Comment (yohamta0) [reply ui/src/features/dashboard/components/__tests__/DashboardTimechart.test.tsx:103]: Fixed in 66b86ff9. The test now freezes the clock, asserts the unfinished run ends at that exact current-time value, and restores real timers after each test. The full frontend suite, typecheck, ESLint, and Prettier checks pass.
